### DISCUSSION TITLE
Using React Query to query and mutate smart contracts

### DISCUSSION THREAD

**Discussion by aalmada** [wevm/wagmi#1435] [Show and tell]
I published a series of articles on how to use React Query to interact with your own smart contracts.
wagmi also uses React Query. The custom hooks created based on the articles and the wagmi hooks complement each other nicely.
I think wagmi users might be interested in reading these: https://antao-almada.medium.com/using-react-query-to-query-smart-contracts-part-1-b7fe1a6697

**Comment by tmm** [comment]
Thanks for sharing! Could be cool to add some info how to use [ABIType](https://github.com/wagmi-dev/abitype) along with React Query too (if you are looking for ideas on how to extend to another part :))

**Reply by aalmada** [reply]
Hi @tmm! I did not know about ABIType. I'll have to try using it. Thanks!
